### ISSUE TITLE
docker network option support for deploying sam applications

### ISSUE THREAD

**Issue by awschristou** [aws/aws-toolkit-vscode#374]
This allows users to deploy sam applications and specify a docker network during the build and deploy steps.

Setting should go inside a template's `options` object

```javascript
{
    templates: {
        [ relativePath: string ]: {
            handlers: {
                [ name: string ]: {
                    dockerNetwork?: string
                }
            }
            parameterOverrides: {
                [ key: string ]: string
            }
            options?: {
                dockerNetwork?: string
            }
        }
    }
}
```

Acceptance Criteria:
* Each template configuration has an optional `dockerNetwork ` boolean 
* when sam applications are deployed, the template config drives usage of the `--docker-network` arg when running sam cli
* if the value is missing, the docker network arg is not used
* autocompletion support for the field provides a description (if #369 is not delivered yet, break this into a separate task)

**Comment by LazerFX** [comment]
As this is something that was making our test and development flow more difficult, I've taken the liberty of implementing this small change.  Happy to adjust as per your comments, hope I've not overstepped any boundaries.

**Comment by awschristou** [comment]
This was released in v1.2.0. Closing
